sklearn/cluster/_spectral.py

```diff
@@ -461,9 +483,24 @@ class SpectralClustering(ClusterMixin, BaseEstimator):
         Number of neighbors to use when constructing the affinity matrix using
         the nearest neighbors method. Ignored for ``affinity='rbf'``.
 
-    eigen_tol : float, default=0.0
-        Stopping criterion for eigendecomposition of the Laplacian matrix
-        when ``eigen_solver='arpack'``.
+    eigen_tol : float, default="auto"
+        Stopping criterion for eigendecomposition of the Laplacian matrix.
+        If `eigen_tol="auto"` then the passed tolerance will depend on the
+        `eigen_solver`:
+
+        - If `eigen_solver="arpack"`, then `eigen_tol=0.0`;
+        - If `eigen_solver="lobpcg"` or `eigen_solver="amg"`, then
+          `eigen_tol=1e-5`.
```

Comment:
With the comment in https://github.com/scikit-learn/scikit-learn/pull/23210#discussion_r862938950, we can have `"auto"` mean `eigen_tol=None` for 'lobpcg' and link to SciPy's `lobpcg` docs that describe what `None` means.